pkg/manifests/manifests.go

```diff
@@ -0,0 +1,59 @@
+package manifests
+
+import (
+	_ "embed"
+
+	olmv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var (
+	//go:embed csv.yaml
+	packageServerCSV []byte
+)
+
+type CSVOption func(*olmv1alpha1.ClusterServiceVersion)
+
+// NewPackageServerCSV is responsible for serializing the PackageServer csv.yaml
+// YAML manifest into a populated ClusterServiceVersion Go structure that contains a
+// metadata.namespace that matches the @namespace value.
+func NewPackageServerCSV(opts ...CSVOption) (*olmv1alpha1.ClusterServiceVersion, error) {
+	var csv olmv1alpha1.ClusterServiceVersion
+	if err := yaml.Unmarshal(packageServerCSV, &csv); err != nil {
+		return nil, err
+	}
+
+	for _, opt := range opts {
+		opt(&csv)
+	}
+
+	return &csv, nil
+}
+
+func WithName(name string) CSVOption {
+	return func(csv *olmv1alpha1.ClusterServiceVersion) {
+		csv.Name = name
+	}
+}
```

Comment:
This may be an indication that we need to either continue in this direction for the fields we care about (namely HA-related fields, like `.spec.replicas`), or start gradually migrating towards using a go-template version of the CSV that's templated dynamically according to HA expectations exposed by in-cluster APIs.